org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/settings/XMLValidationSettings.java

```diff
@@ -244,4 +265,23 @@ public boolean equals(Object obj) {
 		return true;
 	}
 
+	@Override
+	public XMLValidationSettings deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
```

Comment:
I would move this serializer  in an another class like `XMLValidationSettingsTypeAdapter  `to avoid polluting the XMLValidationSettings